exporter/importables.go

```diff
@@ -648,8 +648,10 @@ var resourcesMap map[string]importable = map[string]importable{
 		Name: func(d *schema.ResourceData) string {
 			// TODO: if I have 2 users from different domains: test@domain1.com & test@domain2.com - then I'll generate the same name
 			// use another algorithm for name generation, like, just replace non-word characters with '_'
-			s := strings.Split(d.Get("user_name").(string), "@")
-			return s[0]
+			//s := strings.Split(d.Get("user_name").(string), "@")
+			//return s[0]
+			s := d.Get("user_name").(string)
+			return regexp.MustCompile(`[^a-zA-Z0-9]+`).ReplaceAllString(s, "_")
```

Comment:
for performance, it's better to make it a variable that is declared with `var` at the beginning of importables.
Also remove commented out code.
Although, the best case would be to rely on the name fixes declared here: https://github.com/databricks/terraform-provider-databricks/blob/master/exporter/context.go#L87, although we may need to fix this problematic line: https://github.com/databricks/terraform-provider-databricks/blob/master/exporter/context.go#L91 - it will fix other issues as well

Reply:
NameFixes are applied to **every** name.
That can have other effects across all names. 
Despite the NameFixes array of substitutions we split the databricks_user hcl name then return the user portion.

Reply:
What I meant is just return `d.Get("user_name").(string)` and allow `NameFixes` to handle it...

Reply:
@rohitnijhawan @rohitkn okay, `NameFixes` is a less invasive approach, but still make the full-email-resource-name as opt-in feature, as you're changing the behavior.